src/components/MovieList/MovieList.jsx

```diff
@@ -15,12 +23,25 @@ class MovieList extends Component {
     };
   }
 
-  componentDidMount() {
-    API.getMovies().then((movies) => {
-      this.setState({
-        movies: [...movies.data],
+  componentWillReceiveProps({ location }) {
+    if (/search/.test(location.pathname)) {
+      const nextSearchParams = new URLSearchParams(location.search);
+      const prevSearchParams = new URLSearchParams(this.props.location.search);
+      if (location.pathname === this.props.location.pathname &&
+        nextSearchParams.get('searchBy') === prevSearchParams.get('searchBy') &&
+        nextSearchParams.get('searchFor') === prevSearchParams.get('searchFor')) {
+        console.log('no update');
+        return false;
+      }
+
+      MovieList.getData().then((res) => {
```

Comment:
async can be used here and let's move it to the separate method

Reply:
need some clarifications